### ISSUE TITLE
Flags should not be storing context in the static reference. Potential to m/m leaks!

### ISSUE THREAD

**Issue by whytocode** [jsramraj/flags#1]
Hey @jsramraj, 
Long time no see. Just got to see this library. Great attempt to solve this problem!
I am raising one issue which I saw while skimming through the code.
https://github.com/jsramraj/flags/blob/5ba586bb30ac1e1ed4dc4b9c8493150050609eb5/flags/src/main/java/com/jsramraj/flags/Flags.java#L22

1. Instead of storing the activity context, it should passed into the method calls every time, if it relies on Activity context.
2. If it relies of application context, then we can use this approach but the caveat is you will not be able to take the benefits of localisation, application contexts do not give you localised strings, they will fallback to english.

Solving the existing code, requires something like a builder.
`Flags.with(context).forCountry()`

To make it even better and testable, separate out the user API `forCountry` in some 
```
interface FlagDrawableProvider {
   public BitmapDrawable forCountry(enum or @StringDef countryCode);
}

```
When, user calls `with()`, library returns a concrete implementation of `FlagDrawableProvider`ie Flags. This makes sure that user is not able to access any other method apart from API and we can get rid of all static keywords in the methods, and you can mock them in unit tests.
and if context is `null` or not an instance of `AppCompatActivity`, library can throw an exception.

Edit:
```
public static FlagDrawableProvider with(@NonNull final Context ctx) {'
        if(!ctx instanceof AppCompatActivity ) {
              throw new IllegalArgumentException("Some Message");
         }
        return new Flags();      // In real time, you might want to initialise only for the first time and for 
                                              // next calls reuse the same Flags instance.
    }
```

**Comment by jsramraj** [comment]
@whytocode Good catch, didn't think through that. I'll address it. Thanks.

**Comment by jsramraj** [comment]
This has been addressed in the commit 710bd2f
